src/GNDStk/Node/src/child.hpp

```diff
@@ -0,0 +1,81 @@
+
+// -----------------------------------------------------------------------------
+// Node::child(string)
+//
+// Searches the Node's children for a child of the given name. If found,
+// returns the child node in question. Else, returns an empty Node.
+// -----------------------------------------------------------------------------
+
+// const
+const Node &child(
+   const std::string &key,
+   bool &found = detail::default_bool
+) const {
+   // search
+   for (auto &c : children)
+      if (c != nullptr and c->name == key)
+         return found = true, *c;
```

Comment:
Hence my remark a bit earlier in the same code. I'm building a system where there's a better way to access data than by using simple strings; and the question of "is there one?" versus "are there many?" will be handled much more elegantly. Even with the string key version, I may, as commented about in the code, either reformulate these simple string-key functions, or provide alternative versions to do different things. I've been focusing on better options, but suppose it would be a good idea at this point to go ahead and fill in the blanks here.